### DISCUSSION TITLE
CustomAction WixQuietExec copy file

### DISCUSSION THREAD

**Discussion by sigster** [wixtoolset/issues#8771] [Questions]
Hi
I am try to copy file this don't work
Can anyone help me to get this to work

Regards
Sigster

`		
		Property Id="WixQuietExecCmdLine" Value='"[System64Folder]cmd.exe /c copy [FolderContents]SHPImport_ZW.dll C:\map\SHPImport_ZW.dll"'/>

		
		<CustomAction Id="Copyfile" BinaryKey="WixCA" DllEntry="WixQuietExec" Execute="immediate" Return="ignore"/>

		
		<InstallExecuteSequence>
			<Custom Action='Copyfile' Before='InstallValidate'></Custom>
		</InstallExecuteSequence>`

`
Action start 13:33:15: Copyfile.
MSI (s) (08:90) [13:33:15:957]: Invoking remote custom action. DLL: C:\WINDOWS\Installer\MSID884.tmp, Entrypoint: WixQuietExec
MSI (s) (08:18) [13:33:15:957]: Generating random cookie.
MSI (s) (08:18) [13:33:15:963]: Created Custom Action Server with PID 18216 (0x4728).
MSI (s) (08:20) [13:33:15:990]: Running as a service.
MSI (s) (08:20) [13:33:15:993]: Hello, I'm your 32bit Impersonated custom action server.
MSI (s) (08!88) [13:33:15:999]: PROPERTY CHANGE: Deleting WixQuietExecCmdLine property. Its current value is '"[System64Folder]cmd.exe /c copy [FolderContents]SHPImport_ZW.dll C:\kort\SHPImport_ZW.dll"'.
WixQuietExec:  Error 0x80070002: Command failed to execute.
WixQuietExec:  Error 0x80070002: QuietExec Failed
WixQuietExec:  Error 0x80070002: Failed in ExecCommon method
CustomAction Copyfile returned actual error code 1603 but will be translated to success due to continue marking

`

**Comment by bevanweiss** [comment]
When you write an installer, try to avoid dodgey command line hacks.

If you want to copy a file, then the questions are:
1. Is it REALLY a copy... or is it an install of a file from the MSI to the target computer?  If so, just author it as an InstallFile.
2. If it's a copy of an existing file around the target computer, try https://wixtoolset.org/docs/schema/wxs/copyfile/
